store/root/store.go

```diff
@@ -329,12 +316,27 @@ func (s *Store) Commit(cs *store.Changeset) ([]byte, error) {
 
 	s.workingHash = nil
 
-	// prune SS and SC
-	s.pruningManager.Prune(version)
-
 	return s.lastCommitInfo.Hash(), nil
 }
 
+// Prune prunes the root store to the provided version.
+func (s *Store) Prune(version uint64) error {
+	if s.telemetry != nil {
+		now := time.Now()
+		s.telemetry.MeasureSince(now, "root_store", "prune")
```

Comment:
This doesn't make sense, did you mean to place it in a `defer` block so that we measure **after** pruning is completed?

Reply:
No, this is correct. We provide the time when the function starts. If we `defer` we get the time when the function ends.

Reply:
ohhhh wait, you mean the `MeasureSince` call, not the `time.Now()` call. Yes, you're correct.

@cool-develope the `MeasureSince` needs to be deferred.

Reply:
then, it should be applied everywhere of `MeasureSince`

Reply:
The correct code should read:

```go
		now := time.Now()
		defer s.telemetry.MeasureSince(now, "root_store", "prune")
```

Where `time.Now()` is at the top of the function, and the defer closing over `now` to run when the function returns.